spec/services/teacher_training_api/import_course_spec.rb

```diff
@@ -20,40 +31,96 @@ module TeacherTrainingApi
 
           it "create a course with the correct code and name" do
             subject
-            expect(provider.courses.find_by(code: code, name: name)).to_not be_nil
+            expect(provider.courses.find_by(code: imported_code, name: imported_name)).to_not be_nil
+          end
+
+          it "creates the course with the correct subjects" do
+            expect { subject }.to change { CourseSubject.count }.from(0).to(1)
+            expect(provider.courses.find_by(code: imported_code, name: imported_name).subjects).to match course_subjects
           end
         end
 
         context "and it's draft" do
-          let(:course) { JSON(ApiStubs::TeacherTrainingApi.course(state: "draft")) }
+          let(:imported_course) { JSON(ApiStubs::TeacherTrainingApi.course(state: "draft")) }
 
           it "does not create the course" do
             expect { subject }.not_to(change { Course.count })
           end
+
+          it "does not create the course_subject join" do
+            expect { subject }.not_to(change { CourseSubject.count })
+          end
         end
       end
 
       context "when the course code already exists for that provider" do
-        context "with the same name" do
-          before { create(:course, code: code, name: name, provider: provider) }
+        let(:existing_course) do
+          create(:course,
+                 code: imported_code,
+                 provider: provider,
 
-          it "does not creates a duplicate course" do
-            expect { subject }.not_to(change { Course.count })
+                 name: course_name,
+                 subjects: course_subjects)
+        end
+
+        let(:course_name) { imported_name }
+        before { existing_course }
+
+        context "imported course attributes was not changed" do
```

Comment:
It's quite nice when these `context` string, 'string' together to form a sentence.

Before, the test output read like this:

```
"when the course code already exists for that provider" -> "with the same name" -> it "does not creates a duplicate course"
```

Now it reads like:

```
"when the course code already exists for that provider" -> "imported course attributes was not changed" -> "with the same name" -> it "does not creates a duplicate course"
```

So it's lost its readability.

What are we trying to achieve with these new contexts? They don't wrap any new `lets`. As such, I'd be inclined to remove `context "imported course attributes was not changed"` and `context "imported course attributes was changed"`

Reply:
Fixed, Monday was a Non-English day :sweat_smile: